### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by amr-mokhtar**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+* Ensuring that any code submitted is of the appropriate quality.
+* Ensuring that any code submitted is accompanied with appropriate documentation.
+* Ensuring that any code submitted has been reviewed by peers.
+* Ensuring that the relevant documentation is kept up to date with the code.
+* Ensuring that any identified bugs in the relevant code are captured in the issues backlog.
+* Ensuring that the unit, integration, and regression tests are appropriate for the relevant components.
+* Ensuring that the contribution does not infringe others' Intellectual Property or the appropriate license and that it is in compliance with [Developer Certificate of Origin](http://developercertificate.org/).
+* Answering questions/emails on the relevant areas of the OpenNESS code base.
+
+## Technical Steering Committee
```
Comment (krishnajs) [inline CONTRIBUTING.md:36]: Do we need to say anything about the OpenNESS publishing the roadmap?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+Signing off the contribution is performed by the command `git commit --signoff` which certifies that you wrote it or otherwise have the right to pass it on as an open-source contribution.
+
+Application contributors have the implicit commitment to continually support their applications in whenever a bug is discovered or a test is broken due to the publication of a new release.
+
+It is always admired when application contributors provide end-to-end acceptance test methods.
+
+Big contributions, ideas or controversial features should be discussed with the developers' community and with the Technical Steering Committee through a Request for Comments (RFC). This is an essential early step to bring everybody to a common level of understanding so that when the final implementation is pushed, there will be common consensus of the acceptability of the contribution and smooth integration of it to the mainline. Requests for Comments should be submitted through a pull request prefixed with `[RFC]` in the title.
+
+## Contribution Acceptance Flow
```
Comment (krishnajs) [inline CONTRIBUTING.md:64]: do we need to cover 
1. Unattended PRs will be deleted after a while? 
2. what is our position on someone enabling competition hardware e.g. Mellanox NIC, NVidia GPU etc ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+3. The contributor resolves all findings reported by CI/CD verifications.
+4. The contributor invites reviewers to the pull request and addresses their comments/feedback.
+5. Every pull request must receive 2 approvals in order to go forward with merging.
+6. The maintainer(s) must review and ensure that the change(s) introduced by the pull request meets the acceptance criteria.
+6. Merging the pull requests can only be performed by the maintainer(s).
+
+## How to report an issue/bug/enhancement
+It is encouraged to use the [GitHub Issues](https://github.com/open-ness/edgeapps/issues) tool to report any bug, issue, enhancement or to seek help.
+
+## Resources
```
Comment (krishnajs) [inline CONTRIBUTING.md:76]: what will be the impact of Enhanced and Open on this? How do we make the ref. doc available ? 

Are we saying all the app developers will be Enhanced Openness customers?

File: CONTRIBUTING.md (hunk 1 above)
Comment (amr-mokhtar) [reply CONTRIBUTING.md:36]: Roadmap will be required if we have releasing schedule. For edgeapps there is no release schedule is defined.

File: CONTRIBUTING.md (hunk 2 above)
Comment (amr-mokhtar) [reply CONTRIBUTING.md:64]: 1. It is the responsibility of PR issuers to track and ask for community's reviews. If PR stayed for extended periods of time, the maintainers have the right to close obsolete/inactive PRs.
2. As long as these contributions can be maintained at the long run, they will be accepted. Contributors are responsible to maintain and continuously validate their hardware. Hit & run contributions may risk getting removed from the project.

File: CONTRIBUTING.md (hunk 3 above)
Comment (amr-mokhtar) [reply CONTRIBUTING.md:76]: Yes. The intention is to bring-up contributors up-to-speed with the OpenNESS solution.

**Pull request merged**